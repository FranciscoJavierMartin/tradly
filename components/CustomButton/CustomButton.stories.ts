import CustomButton from './CustomButton.vue';

export default {
  title: 'CustomButton',
  component: CustomButton,
  args: {
    text: 'Text',
  },
};

const Template = (args) => ({
  components: { CustomButton },
  setup() {
    return { args };
  },
  template: '<CustomButton v-bind="args"/>',
});

export const Basic = Template.bind({});
Basic.args = {};

export const Inverted = Template.bind({});
Inverted.args = {
  inverted: true,
};
