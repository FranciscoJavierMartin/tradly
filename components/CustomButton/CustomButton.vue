<template>
  <button
    class="c-custom-button"
    :class="{ 'c-custom-button--inverted': inverted }"
    @click="onClick"
  >
    {{ text }}
  </button>
</template>

<script setup lang="ts">
import { PropType } from 'vue';

const props = defineProps({
  text: {
    type: String,
    required: true,
  },
  onClick: {
    type: Function as PropType<(payload: MouseEvent) => void>,
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    default: (payload: MouseEvent) => {},
  },
  inverted: {
    type: Boolean,
    required: false,
  },
});
</script>

<style lang="scss">
@import '../../assets/styles/main.scss';

.c-custom-button {
  background-color: $dark-green;
  border-radius: 24px;
  border: none;
  color: $white;
  font-family: $monserrat-font;
  font-size: 18px;
  font-weight: 600;
  line-height: 18px;
  padding: 15px;
  text-align: center;
  text-decoration: none;
  width: 100%;

  &--inverted {
    background-color: $white;
    color: $light-green;
  }
}
</style>
