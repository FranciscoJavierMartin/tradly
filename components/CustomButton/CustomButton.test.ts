import '@testing-library/jest-dom';
import { describe, expect, test } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/vue';
import CustomButton from './CustomButton.vue';

describe('CustomButton.vue', () => {
  test('should render', () => {
    const { container } = render(CustomButton, {
      props: {
        text: 'Lorem ipsum',
      },
    });

    expect(
      screen.getByRole('button', { name: /Lorem ipsum/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should call the function passed when click on the button', async () => {
    const mockFunction = vi.fn();
    render(CustomButton, {
      props: {
        text: 'Lorem ipsum',
        onClick: mockFunction,
      },
    });

    await fireEvent.click(screen.getByText('Lorem ipsum'));
    expect(mockFunction).toHaveBeenCalledOnce();
  });
});
