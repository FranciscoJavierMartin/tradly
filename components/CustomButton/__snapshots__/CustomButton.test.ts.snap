// Vitest Snapshot v1

exports[`CustomButton.vue > should render 1`] = `
<button
  class="c-custom-button"
>
  Lorem ipsum
</button>
`;
